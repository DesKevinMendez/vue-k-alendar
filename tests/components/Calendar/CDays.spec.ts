
import CDays from '@/components/Calendar/CDays.vue';
import useConfig from '@/composables/useConfig';
import { mount, VueWrapper } from '@vue/test-utils';
const { setLang } = useConfig()

let wrapper: VueWrapper<InstanceType<typeof CDays>>

describe('CDays', () => {

  beforeEach(() => {
    wrapper = mount(CDays);
  });

  afterEach(() => {
    vi.clearAllMocks()
  })

  it('should render the component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('lang', () => {
    it('by default is en', async () => {
      const days = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun',];
      const container = wrapper.find('.k-alendar-days-container');

      container.findAll('div')
        .forEach((day, index) => {
          expect(day.text()).toBe(days[index]);
        });
    })

    it('is es', async () => {
      setLang('es');
      await wrapper.vm.$nextTick();

      const days = ['lun', 'mar', 'mié', 'jue', 'vie', 'sáb', 'dom',];
      const container = wrapper.find('.k-alendar-days-container');

      container.findAll('div')
        .forEach((day, index) => {
          expect(day.text()).toBe(days[index]);
        });
    })

    it('is it', async () => {
      setLang('it');
      await wrapper.vm.$nextTick();

      const days = ['lun', 'mar', 'mer', 'gio', 'ven', 'sab', 'dom',];
      const container = wrapper.find('.k-alendar-days-container');

      container.findAll('div')
        .forEach((day, index) => {
          expect(day.text()).toBe(days[index]);
        });
    })

    it('lang is invalid, it should be english', async () => {
      setLang('invalid');
      await wrapper.vm.$nextTick();

      const days = ['lun', 'mar', 'mié', 'jue', 'vie', 'sáb', 'dom',];
      const container = wrapper.find('.k-alendar-days-container');

      container.findAll('div')
        .forEach((day, index) => {
          expect(day.text()).toBe(days[index]);
        });
    })
  })
});