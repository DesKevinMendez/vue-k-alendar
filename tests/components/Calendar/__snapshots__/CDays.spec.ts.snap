// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CDays > should render the component 1`] = `
"<div data-v-a8a4dcb5="" class="k-alendar-days-container">
  <div data-v-a8a4dcb5="">Mon</div>
  <div data-v-a8a4dcb5="">Tue</div>
  <div data-v-a8a4dcb5="">Wed</div>
  <div data-v-a8a4dcb5="">Thu</div>
  <div data-v-a8a4dcb5="">Fri</div>
  <div data-v-a8a4dcb5="">Sat</div>
  <div data-v-a8a4dcb5="">Sun</div>
</div>"
`;
